File: backend/TenderGo.Models/Requests/BidInsertRequest.cs
using System.ComponentModel.DataAnnotations;

public class BidInsertRequest
{
    [Required]
    [Range(0.01, double.MaxValue)]
    public decimal Price { get; set; }

    [Required]
    public int TenderId { get; set; }

    [StringLength(500)]
    public string? Note { get; set; }

    [Required]
    public string UserId { get; set; } = string.Empty;

    [Required]
    [Range(1, int.MaxValue)]
    public int DeliveryDays { get; set; }
}

File: UI/lib/shared/models/requests/bid_insert_request.dart
class BidInsertRequest {
  final double price;
  final int tenderId;
  final String? note;
	final String userId;
  final int deliveryDays;

  BidInsertRequest({
    required this.price,
    required this.tenderId,
		required this.userId,
    this.note,
    required this.deliveryDays,
  });

  Map<String, dynamic> toJson()=> {
      'price': price,
      'tenderId': tenderId,
      'note': note,
      'userId': userId,
      'deliveryDays': deliveryDays,
  };
}